src/Form.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import { fetchData } from "./Api.jsx";
+import ResponseContent from "./ResponseContent.jsx";
+const Form = () => {
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [response, setResponse] = useState(null);
+  const [loading, setLoading] = useState(null);
+
+  const submitForm = async (e) => {
+    e.preventDefault();
+
+    setLoading(true);
+    const params = {
+      url: "https://httpbin.org/post",
+      method: "post",
+      data: { name, password },
+    };
+    setResponse(await fetchData(params));
+    setLoading(false);
+  };
+  return (
+    <div>
+      <form onSubmit={submitForm}>
+        <label>
+          Name:
+          <input
+            type="text"
+            value={name}
+            onChange={(e) => setName(e.target.value)}
+          />
+        </label>
+        <br />
+        <label>
+          Password:
+          <input
+            type="password"
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
```

Comment:
Please read this article
https://www.linkedin.com/pulse/handling-events-react-best-practices-common-mistakes-faizan/